MobileBuy/buy/src/main/java/com/shopify/buy/utils/AndroidPayHelper.java

```diff
@@ -0,0 +1,391 @@
+        }
+
+        Address address = new Address();
+
+        String[] nameParts = extractFirstAndLastNames(walletUserAddress.getName());
+        address.setFirstName(nameParts[FIRST_NAME_PART]);
+        address.setLastName(nameParts[LAST_NAME_PART]);
+
+        address.setAddress1(walletUserAddress.getAddress1());
+
+        // Google Wallet address has 5 address lines.
+        // Convert to 2 address lines
+        StringBuilder address2 = new StringBuilder();
+        address2.append(walletUserAddress.getAddress2());
+
+        if (!TextUtils.isEmpty(walletUserAddress.getAddress3())) {
+            address2.append(walletUserAddress.getAddress3());
+            address2.append(", ");
+        }
+        if (!TextUtils.isEmpty(walletUserAddress.getAddress4())) {
+            address2.append(walletUserAddress.getAddress4());
+            address2.append(", ");
+        }
+        if (!TextUtils.isEmpty(walletUserAddress.getAddress4())) {
+            address2.append(walletUserAddress.getAddress4());
+            address2.append(", ");
+        }
+        if (!TextUtils.isEmpty(walletUserAddress.getAddress5())) {
+            address2.append(walletUserAddress.getAddress5());
+        }
+
+        address.setAddress2(address2.toString());
+
+        address.setZip(walletUserAddress.getPostalCode());
+        address.setCity(walletUserAddress.getLocality());
+        address.setCountryCode(walletUserAddress.getCountryCode());
+        address.setProvinceCode(walletUserAddress.getAdministrativeArea());
+        address.setPhone(walletUserAddress.getPhoneNumber());
+        return address;
+    }
+
+    private static String[] extractFirstAndLastNames(String name) {
+        String[] firstAndLastName = new String[2];
+        String[] nameParts = name.split(" ");
+        if (nameParts.length > 0) {
+            firstAndLastName[FIRST_NAME_PART] = (nameParts[FIRST_NAME_PART]);
+        }
+        int nameIndex = 1;
+        String lastName = "";
+        while (nameParts.length > nameIndex) {
+            lastName += nameParts[nameIndex++];
+        }
+        firstAndLastName[LAST_NAME_PART] = lastName.trim();
+        return firstAndLastName;
+    }
+
+    /**
+     * Creates a Masked Wallet Request from a Shopify Checkout
+     *
+     * @param merchantName        The merchant name to show on the Android Pay dialogs, not null or empty
+     * @param checkout            The {@link Checkout} to use, not null.
+     * @param buyClient           The {@link BuyClient} to use, not null.  Must have previously had Android Pay enabled via {@link BuyClient#enableAndroidPay(String)}
+     * @param phoneNumberRequired If true, the phone number will be required as part of the Shipping Address in Android Pay
+     * @return A {@link MaskedWalletRequest}
+     */
+    public static MaskedWalletRequest createMaskedWalletRequest(String merchantName, Checkout checkout, BuyClient buyClient, boolean phoneNumberRequired) {
```

Comment:
I would remove reference to BuyClient and pass instead androidPayPublicKey, it will help to reduce the complexity of dependencies, really we don't need buyClient reference only key so let's pass the it and allow user to decide where this key will be taken.